docs/docs/developers/testing/cheat_codes.md

```diff
@@ -500,13 +500,13 @@ Note: One Field element occupies a storage slot. Hence, structs with multiple fi
 
 ```rust
 struct Storage {
-    balances: Map<AztecAddress, PublicState<Field, FIELD_SERIALIZED_LEN>>,
+    balances: Map<AztecAddress, PublicState<Field>>,
 }
 
 impl Storage {
     fn init() -> Self {
         Storage {
-            balances: Map::new(1, |slot| PublicState::new(slot, FieldSerializationMethods)),
+            balances: Map::new(context, 1, |context, slot| PublicState::new(context, slot)),
```

Comment:
Oh, but I don't see the map.nr file in the set of changes in this PR?

Reply:
Just updated the docs, since the Map API was outdated in this file.

The context is necessary because the inner lambda needs to pass it to the actual storage implementation, don't really think there's a way around it...also it would make https://github.com/AztecProtocol/aztec-packages/pull/4200 a lot more complicated.

Reply:
Map always required context, it wasn't there previously because of a typo I think. Also storage init needs context as param I think

Reply:
There could be a way by using a closure with environment instead of just a lambda function. But since most of the time the init will be autogenerated it might not be very useful and it might complicate the work of the macro that will autogenerate init